test/e2e/specs/specs-i18n/wp-i18n__editor.ts

```diff
@@ -214,85 +214,101 @@ const translations: Translations = {
 		],
 	},
 };
-const locale = BrowserHelper.getLocale() as LanguageSlug;
-const localeTranslations = translations[ locale ];
-const describeSkipNoTranslations = localeTranslations ? describe : describe.skip;
 
-describeSkipNoTranslations(
-	DataHelper.createSuiteTitle( `Editor Translations (${ locale })` ),
-	() => {
-		let gutenbergEditorPage: GutenbergEditorPage;
-		let page: Page;
+describe( 'I18N: Editor', function () {
+	let page: Page;
+	// Filter out the locales that do not have valid translation content defined above.
+	const locales = Object.keys( translations ).filter( ( locale ) =>
+		TestEnvironment.LOCALES().includes( locale )
+	);
 
-		setupHooks( ( args ) => {
-			page = args.page;
-
-			// Confirm page leave with unsaved changes prompt.
-			page.on( 'dialog', async ( dialog ) => {
-				if ( dialog.type() === 'beforeunload' ) {
-					await dialog.accept();
-				}
-			} );
-		} );
-
-		it( 'Log in', async () => {
-			const loginPage = new LoginPage( page );
-			await loginPage.login( { account: 'i18nUser' } );
+	setupHooks( ( args ) => {
+		page = args.page;
+		// Confirm page leave with unsaved changes prompt.
+		page.on( 'dialog', async ( dialog ) => {
+			if ( dialog.type() === 'beforeunload' ) {
+				await dialog.accept();
+			}
 		} );
+	} );
 
-		it( 'Change UI language', async () => {
-			await Promise.all( [
-				page.waitForNavigation( { url: '**/home/**', waitUntil: 'load' } ),
-				page.goto( DataHelper.getCalypsoURL( '/' ) ),
-			] );
+	it( 'Log in', async function () {
+		const loginPage = new LoginPage( page );
+		await loginPage.login( { account: 'i18nUser' } );
+	} );
 
-			const changeUILanguageFlow = new ChangeUILanguageFlow( page );
-			await changeUILanguageFlow.changeUILanguage( locale );
+	describe.each( locales )( `Locale: %s`, function ( locale ) {
```

Comment:
Parametrize the setup, start new post and block verification for each language that has translation defined.